mr/src/main/java/org/apache/iceberg/mr/InputFormatConfig.java

```diff
@@ -85,6 +87,14 @@ public ConfigBuilder(Configuration conf) {
     }
 
     public Configuration conf() {
+      // Store the io and the current snapshot of the table in the configuration which are needed for the split
+      // generation
+      Table table = Catalogs.loadTable(conf);
+
+      // The FileIO serializes the configuration and we might end up recursively serializing the objects.
+      // To avoid this unset the value before serialization and set it again in the next line.
+      conf.unset(InputFormatConfig.FILE_IO);
+      conf.set(InputFormatConfig.FILE_IO, SerializationUtil.serializeToBase64(table.io()));
```

Comment:
Shouldn't everything use the `FileIO` from the table instead of from the config?